nvflare/fuel/hci/cmd_arg_utils.py

```diff
@@ -68,3 +71,82 @@ def get_usage(self) -> str:
         usage_output = buffer.getvalue().split("\n", 1)[1]
         buffer.close()
         return usage_output
+
+
+def process_targets_into_str(targets: List[str]) -> str:
+    if not isinstance(targets, list):
+        raise SyntaxError("targets is not a list.")
+    if not all(isinstance(t, str) for t in targets):
+        raise SyntaxError("all targets in the list of targets must be strings.")
+    for t in targets:
+        try:
+            validate_required_target_string(t)
+        except SyntaxError:
+            raise SyntaxError("each target in targets must be a string of only valid characters and no spaces.")
+    return " ".join(targets)
+
+
+def validate_required_target_string(target: str) -> str:
+    """Returns the target string if it exists and is valid."""
+    if not target:
+        raise SyntaxError("target is required but not specified.")
+    if not isinstance(target, str):
+        raise SyntaxError("target is not str.")
+    if not re.match("^[A-Za-z0-9._-]*$", target):
+        raise SyntaxError("target must be a string of only valid characters and no spaces.")
+    return target
+
+
+def validate_options_string(options: str) -> str:
+    """Returns the options string if it is valid."""
+    if not isinstance(options, str):
+        raise SyntaxError("options is not str.")
+    if not re.match("^[A-Za-z0-9- ]*$", options):
+        raise SyntaxError("options must be a string of only valid characters.")
+    return options
+
+
+def validate_path_string(path: str) -> str:
+    """Returns the path string if it is valid."""
+    if not isinstance(path, str):
+        raise SyntaxError("path is not str.")
+    if not re.match("^[A-Za-z0-9-._/]*$", path):
+        raise SyntaxError("unsupported characters in path {}".format(path))
+    if path.startswith("/"):
+        raise SyntaxError("absolute path is not allowed")
+    paths = path.split("/")
+    for p in paths:
+        if p == "..":
+            raise SyntaxError(".. in path name is not allowed")
+    return path
+
+
+def validate_file_string(file: str) -> str:
+    """Returns the file string if it is valid."""
+    if not isinstance(file, str):
+        raise SyntaxError("file is not str.")
+    if not re.match("^[A-Za-z0-9-._/]*$", file):
+        raise SyntaxError("unsupported characters in file {}".format(file))
+    if file.startswith("/"):
+        raise SyntaxError("absolute path for file is not allowed")
+    paths = file.split("/")
+    for p in paths:
+        if p == "..":
+            raise SyntaxError(".. in file path is not allowed")
```

Comment:
since this part duplicated with "validate_path_string" can we reuse that method here?

Reply:
Probably so.

Reply:
done as suggested